Il2CppInspector.Common/Reflection/MethodBase.cs

```diff
@@ -110,6 +110,19 @@ public abstract class MethodBase : MemberInfo
 
             // Regular method
             : Name;
+        //for get rid of compiler generated characters when generate c# code 
+        public new string CSharpSafeName
+        {
+            get
+            {
+                var index = CSharpName.LastIndexOf(".");
+                return (index >= 0)
+                        //implement for interface may contains generic parameters
+                        ? (CSharpName.Substring(0, index + 1) + CSharpName.Substring(index + 1).Replace("<", "_").Replace(">", "_"))
+                        : CSharpName.Replace("<", "_").Replace(">", "_");
+
+            }
+        }
```

Comment:
There is no need to create a new property here as we will end up with `CSharpName` and `CSharpSafeName`, we only need one property to correctly return the C# name of a method because there is only one possible correct name for a given method. You can add this code to the end of `MethodBase.CSharpName`.

But, will this not mess up generic method names, eg `void Foo<T>(T bla)`?